src/Module/Notifications/Notification.php

```diff
@@ -107,24 +108,31 @@ public static function content(array $parameters = [])
 		$request_id = $parameters['id'] ?? false;
 
 		if ($request_id) {
-			$notify = DI::notify()->getByID($request_id, local_user());
+			$Notify = DI::notify()->selectOneById($request_id);
+			if ($Notify->uid !== local_user()) {
+				throw new HTTPException\ForbiddenException();
+			}
 
 			if (DI::pConfig()->get(local_user(), 'system', 'detailed_notif')) {
-				$notify->seen = true;
-				DI::notify()->update($notify);
+				$Notify->setSeen();
+				DI::notify()->save($Notify);
 			} else {
-				DI::notify()->setSeen(true, $notify);
+				if ($Notify->uriId) {
+					DI::notification()->setAllSeenForUser($Notify->uid, ['target-uri-id' => $Notify->uriId]);
+				}
+
+				DI::notify()->setAllSeenForRelatedNotify($Notify);
 			}
 
-			if (!empty($notify->link)) {
-				System::externalRedirect($notify->link);
+			if ($Notify->link) {
+				System::externalRedirect($Notify->link);
 			}
 
 			DI::baseUrl()->redirect();
 		}
 
 		DI::baseUrl()->redirect('notifications/system');
 
-		throw new HTTPException\InternalServerErrorException('Invalid situation.');
+		return '';
```

Comment:
Why did you delete the Exception? I thought this point "should never happen", so an exception would be appropriate?

Reply:
Both of these lines, the one that was removed and the one I added, are only "needed" to make static analyzers happy since they don't know that `DI::baseUrl()->redirect` interrupts the function flow. Neither of them will ever be called because of the `exit;` in `redirect`, both lines prevent an annoying `Missing return statement` message while the Exception throwing prompts an additional `Missing @throws tag in method documentation` message. In the end, the `return` statement makes the file one line shorter and that's why I went with it. 🤷‍♂️

Reply:
A I see, just a quick idea/note for "later": the clean way would be that we don't call redirects directly and try to route per Routing class (Requesr -> Response), so we would have one point where the process stops and not that many :)